forte/solvers/solver.py

```diff
@@ -0,0 +1,119 @@
+from abc import ABC, abstractmethod
+import logging
+
+from forte.data import Data
+from forte.model import MolecularModel
+from forte.results import Results
+from forte.molecule import Molecule
+from forte.basis import Basis
+
+import forte
+
+
+class Solver(ABC):
+    """
+    A class used to implement a quantum chemistry solver.
+
+    Solver stores Forte base objects in a data attribute
+    and a results object.
+    """
+    def __init__(self):
+        self._executed = False
+        self._data = Data()
+        self._results = Results()
+        self._output_file = 'output.dat'
+
+    @abstractmethod
+    def run(self):
+        """Interface for running the solver."""
+
+    @property
+    def results(self):
+        return self._results
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def executed(self):
+        return self._executed
+
+    @property
+    def psi_wfn(self):
+        return self.data.psi_wfn
+
+    @psi_wfn.setter
+    def psi_wfn(self, val):
+        self.data.psi_wfn = val
+
+    @property
+    def scf_info(self):
+        return self.data.scf_info
+
+    @scf_info.setter
+    def scf_info(self, val):
+        self.data.scf_info = val
+
+    @property
+    def mo_space_info(self):
+        return self.data.mo_space_info
+
+    @mo_space_info.setter
+    def mo_space_info(self, val):
+        self.data.mo_space_info = val
+
+    @property
+    def model(self):
+        return self.data.model
+
+    @property
+    def output_file(self):
+        return self._output_file
+
+    @output_file.setter
+    def output_file(self, val):
+        self._output_file = val
+
+    def value(self, label):
+        return self._results.value(label)
+
+    def make_mo_space_info(self, mo_spaces):
+        nmopi = self.scf_info.nmopi()
+        return forte.make_mo_space_info_from_map(nmopi, self.model.point_group, mo_spaces)
+
+    def state(self, charge: int, multiplicity: int, ms: float = None, sym: str = None):
+        return self.data.model.state(charge, multiplicity, ms, sym)
+
+
+class BasicSolver(Solver):
```

Comment:
What's the purpose of this class?

Reply:
In the current workflow, this is the equivalent to an empty `Wavefunction` object in psi4. It is used later in the code in this way:
```python
    # create an empty solver and pass the model in
    solver = BasicSolver()
    solver.data.model = MolecularModel(
        molecule=molecule, basis=basis, scf_aux_basis=scf_aux_basis, corr_aux_basis=corr_aux_basis
    )
```

Reply:
I forgot that you can't instantiate an ABC. This makes sense.